routes/project/location/postgres.rb

```diff
@@ -538,6 +538,50 @@ class Clover
           end
         end
       end
+
+      r.is "upgrade" do
+        r.get do
+          authorize("Postgres:view", pg.id)
+
+          unless pg.needs_upgrade?
+            raise CloverError.new(400, "InvalidRequest", "Database is not upgrading")
+          end
+
+          {
+            current_version: pg.current_version,
+            desired_version: pg.version,
+            upgrade_status: pg.upgrade_status,
+            upgrade_progress: pg.upgrade_progress
+          }
+        end
+
+        r.post do
+          authorize("Postgres:edit", pg.id)
+
+          Validation.validate_postgres_upgrade(
+            version_int: pg.version_int,
+            needs_convergence: pg.needs_convergence?,
+            ongoing_failover: pg.ongoing_failover?,
+            read_replica: pg.read_replica?,
+            needs_upgrade: pg.needs_upgrade?
+          )
+
+          audit_log(pg, "upgrade")
+          pg.update(version: pg.version_int + 1)
```

Comment:
Typically, `audit_log` call comes after action, and is inside the same transaction:

```suggestion
          DB.transaction do
            pg.update(version: pg.version_int + 1)
            audit_log(pg, "upgrade")
          end
```